model/host/host.go

```diff
@@ -794,6 +794,23 @@ func (h *Host) SetPortMapping(portsMap PortMap) error {
 	return nil
 }
 
+func (h *Host) UpdateCachedDistro(distro distro.Distro) error {
+	err := UpdateOne(
+		bson.M{IdKey: h.Id},
+		bson.M{
+			"$set": bson.M{
+				DistroKey: distro,
```

Comment:
This could just set ProviderSettingsList I think, which may also make more sense than passing in h.Distro, could just pass in the []birch.Document{newSettingsDocument}

Reply:
done